### ISSUE TITLE
Replacing jQuery with "vanilla js" broke compatibility with IE11

### ISSUE THREAD

**Issue by heppstux** [heiseonline/shariff#220]
Not that I'd miss IE11 at all, but unfortunately some of our customers do ;)

```
SCRIPT5007: Unable to get property 'classList' of undefined or null reference
```

Happens in dom.js function DOMQuery.prototype.addClass right at initialising Shariff.

System is Win10 with IE11. Edge does the trick and – very strangely IE11

On win8.1 things get even worse with:
```
Unable to get property 'name' of undefined or null reference
```
In function _addButtonList

...

**Comment by liayn** [comment]
@glaszig Can you take a look, what's going on here?

**Comment by glaszig** [comment]
looks like `classList` and `Array.forEach()` are not available on ie11/win8.1.
i don't have windows at hand right now. will have to download one of the dev images. will take some time.

**Comment by liayn** [comment]
@heppstux says "System is Win10 with IE11". So I suppose it does not depend on the OS version.

**Comment by glaszig** [comment]
>  I suppose it does not depend on the OS version.

that's what you'd think. on win7 build 7601 sp1 + ie 11.0.9600.17420 i don't get any of the above but:
    
    SCRIPT438: Object doesn't support property or method 'assign'

**Comment by glaszig** [comment]
@heppstux are you sure it's ie11? because [`classList`](http://caniuse.com/#feat=classlist) and [`forEach()`](http://caniuse.com/#feat=es5) are both supposed to be supported with ie11.

**Comment by heppstux** [comment]
@glaszig, I'm sure it's IE11 on win10

![screen shot 2017-05-15 at 19 33 38](https://cloud.githubusercontent.com/assets/5698638/26070619/8f23ccf4-39a5-11e7-85e6-a9c5ad36232d.png)

**Comment by glaszig** [comment]
@heppstux thanks. that's the error i fixed in #221. are you able to verify my branch over there?

**Comment by heppstux** [comment]
@glaszig I checked your branch as far as I can see the error on the demo-page (that's what the screenshot is showing) is gone.

Give me some time to check if the original error (missing classList) is still reproducible in my project.

**Comment by heppstux** [comment]
Okay, I see this:
![screen shot 2017-05-16 at 11 56 10](https://cloud.githubusercontent.com/assets/5698638/26100637/bc560a14-3a2e-11e7-9ca3-1855e0440a35.png)

**Comment by heppstux** [comment]
Seems to be this line (shariff.js Line 206):
```javascript
var $buttonList = $('<ul>').addClass(themeClass).addClass(orientationClass).addClass(serviceCountClass);
```
... that causes the error.

**Comment by glaszig** [comment]
cannot reproduce on win8.1 + ie11 and win7 + ie11.
also your console example confuses me. `$` should not be a reference to `dq()` in the global scope. how did you get there?

this is me trying this on win8.1 + ie11:

<img width="187" alt="bildschirmfoto 2017-05-16 um 15 20 03" src="https://cloud.githubusercontent.com/assets/201135/26108032/85ec013c-3a4b-11e7-9e1e-ee4d39168255.png">

**Comment by heppstux** [comment]
I used a breakpoint on the mentioned line and jumped right into the local scope there – thus I have a dom.js' $ there – little debugger magic ;) though IE's is really a pain in the a**

**Comment by glaszig** [comment]
ok. here's what i get (win8.1 + ie11).

<img width="516" alt="bildschirmfoto 2017-05-16 um 18 10 51" src="https://cloud.githubusercontent.com/assets/201135/26116332/11bacc5e-3a63-11e7-926a-5e9c5c4298cf.png">

win10 + ie11 images seem to not be available anymore. cannot test that combination. but it would be really weird if the os makes a difference.

**Comment by glaszig** [comment]
pushed 2 other commits eliminating the use of `ClassList` and `Array.forEach()`. please test.

**Comment by glaszig** [comment]
here's the kicker: shariff works on ie11+win10 for me **unless** i open dev tools. if f12 tools are open, i get "null-pointer" complaints.

**Comment by glaszig** [comment]
this is not funny. 50% my fault, 50% microsoft's fault:

i'm using the following code to create elements on the fly:

```js
var fragment = function (html) {
  var div = document.createElement('div');
  div.innerHTML = html;
  return div.children;
};
```

this returns an `HTMLCollection` which behaves like an array -- it has integer indexes which you use to iterate over the elements in the collection. so its object properties look like this:

```js
var html = fragment('<ul>');
for (var p in html) console.log(p);
```

```
0
length
item
namedItem
```

there's one element in the collection which you can access with `html[0]`.

**now turn on F12 developer tools and suddenly behavior changes.** above code will yield:

```
ms__id187
length
item
namedItem
```

of course, iterating over a `DOMQuery`s internal collection will now fall flat because there's no `this[0]` but `this['ms_id187']`.

now, my fault here was to not convert the `HTMLCollection` into an `Array` to make the `each()` function happy and detect the collection as an array and not an object.

i have fixed that in #221 and it works on my side with f12 activated and without. please verify.

**Comment by heppstux** [comment]
Can confirm – everything fine now. Kudos for figuring that out – 🍰
